### ISSUE TITLE
[BUG] KV Cache INT8 doesn't support model which is  trained by zero2 + QLoRA

### ISSUE THREAD

**Issue by amulil** [InternLM/lmdeploy#801]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.

### Describe the bug

I trained a model using zero2 + qlora.

And I want convert it to `KV Cache INT8`, then there is a error log:
```python
WARNING:root:Some parameters are on the meta device device because they were offloaded to the cpu.
Move model.embed_tokens to GPU.
Traceback (most recent call last):
  File "/data/home//.conda/envs/ld2_v010a1/bin/lmdeploy", line 8, in <module>
    sys.exit(run())
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/lmdeploy/cli/cli.py", line 139, in run
    fire.Fire(cli, name='lmdeploy')
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/lmdeploy/cli/lite.py", line 57, in calibrate
    calibrate(model,
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/lmdeploy/lite/apis/calibrate.py", line 155, in calibrate
    _prepare_for_calibrate(model, layer_type, 'lm_head', device)
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/lmdeploy/lite/apis/calibrate.py", line 94, in _prepare_for_calibrate
    _prepare_for_calibrate(child, layer_type, head_name, device,
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/lmdeploy/lite/apis/calibrate.py", line 94, in _prepare_for_calibrate
    _prepare_for_calibrate(child, layer_type, head_name, device,
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/lmdeploy/lite/apis/calibrate.py", line 91, in _prepare_for_calibrate
    child.to('cpu')
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1160, in to
    return self._apply(convert)
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply
    module._apply(fn)
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply
    module._apply(fn)
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/torch/nn/modules/module.py", line 833, in _apply
    param_applied = fn(param)
  File "/data/home//.conda/envs/ld2_v010a1/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1158, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
NotImplementedError: Cannot copy out of meta tensor; no data!
```

### Reproduction

```
lmdeploy lite calibrate --model /path/to/model_trained_by_zero2_and_qlora --calib_dataset 'c4' --calib_samples 128 --calib_seqlen 2048 -work_dir ./kvcache-int8
```

### Environment

```Shell
sys.platform: linux
Python: 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0,1,2,3,4,5,6,7: NVIDIA A100-SXM4-40GB
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 12.1, V12.1.66
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
PyTorch: 2.1.1+cu121
PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v3.1.1 (Git Hash 64f6bcbcbab628e96f33a62c3e975f8535a7bde4)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 12.1
  - NVCC architecture flags: -gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_90,code=sm_90
  - CuDNN 8.9.2
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=12.1, CUDNN_VERSION=8.9.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-unused-private-field -Wno-aligned-allocation-unavailable -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.1.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

LMDeploy: 0.1.0a1+
transformers: 4.35.2
gradio: 3.50.2
fastapi: 0.104.1
pydantic: 2.5.2
```

**Comment by amulil** [comment]
> Can this model be used for inference with lmdeploy without quantization?

Yes, I can use it by `lmdeploy serve api_server `. @pppppM

**Comment by pppppM** [comment]
What is the base model used for finetune? Is it possible to upload the trained model to the Hugging Face Hub? That way I would be able to reproduce your issue faster.

**Comment by SingJumpRapp** [comment]
I had the same problem when I do AWQ quantization.

**Comment by pppppM** [comment]
I have been unable to reproduce this issue. 
Could you provide your environment details? @SingJumpRapp @amulil

**Comment by SingJumpRapp** [comment]
> I have been unable to reproduce this issue. Could you provide your environment details? @SingJumpRapp @amulil
I use four nvidia A6000 to deploy my llama2 70B model, and I can use it by "lmdeploy serve api_server". But when I do AWQ quantization I will meet the error. (lmdeploy lite calibrate --model $HF_MODEL --calib_dataset c4 --calib_samples 128 --calib_seqlen 2048 --work_dir ./quant/ )

The following is my specific environment：
cuda version: cuda12.0
gpu0,1,2,3：NVIDIA A6000
model: HF  llama2-70B
python: python3.10


accelerate               0.25.0
addict                   2.4.0
annotated-types          0.6.0
anyio                    3.7.1
certifi                  2023.11.17
charset-normalizer       3.3.2
click                    8.1.7
cmake                    3.27.9
exceptiongroup           1.2.0
fastapi                  0.104.1
filelock                 3.13.1
fire                     0.5.0
fsspec                   2023.12.0
h11                      0.14.0
huggingface-hub          0.19.4
idna                     3.6
importlib-metadata       7.0.0
Jinja2                   3.1.2
lit                      17.0.6
lmdeploy                 0.1.0a1
markdown-it-py           3.0.0
MarkupSafe               2.1.3
mdurl                    0.1.2
mmengine-lite            0.10.1
mpmath                   1.3.0
networkx                 3.2.1
numpy                    1.26.2
nvidia-cublas-cu11       11.10.3.66
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu11   11.7.101
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu11        8.5.0.96
nvidia-cudnn-cu12        8.9.2.26
nvidia-cufft-cu11        10.9.0.58
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu11       10.2.10.91
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu11     11.4.0.1
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu11     11.7.4.91
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu11         2.14.3
nvidia-nccl-cu12         2.18.1
nvidia-nvjitlink-cu12    12.3.101
nvidia-nvtx-cu11         11.7.91
nvidia-nvtx-cu12         12.1.105
packaging                23.2
pip                      23.3.1
platformdirs             4.1.0
psutil                   5.9.6
pydantic                 2.5.2
pydantic_core            2.14.5
Pygments                 2.17.2
PyYAML                   6.0.1
regex                    2023.10.3
requests                 2.31.0
rich                     13.7.0
safetensors              0.4.1
sentencepiece            0.1.99
setuptools               68.0.0
shortuuid                1.0.11
six                      1.16.0
sniffio                  1.3.0
starlette                0.27.0
sympy                    1.12
termcolor                2.4.0
tiktoken                 0.5.2
tokenizers               0.15.0
tomli                    2.0.1
torch                    2.0.1
tqdm                     4.66.1
transformers             4.35.2
triton                   2.0.0
typing_extensions        4.8.0
urllib3                  2.1.0
uvicorn                  0.24.0.post1
wheel                    0.41.2
yapf                     0.40.2
zipp                     3.17.0

**Comment by pppppM** [comment]
The issue is fixed in  #848 @amulil @SingJumpRapp
